subscriptions/providers/apple_in_app/__init__.py

```diff
@@ -289,12 +319,16 @@ def _handle_refund(self, notification: AppStoreNotification) -> None:
                            transaction_info.original_transaction_id)
             refunded_payment = self._get_latest_transaction(transaction_info.original_transaction_id)
 
+        assert refunded_payment, f'Refund received for {transaction_info=} where no payments exist.'
+
         refunded_payment.subscription.end = transaction_info.revocation_date
         refunded_payment.subscription.save()
         refunded_payment.subscription_end = transaction_info.revocation_date
+        refunded_payment.status = SubscriptionPayment.Status.CANCELLED
+
         refunded_payment.save()
 
-    @transaction.atomic
+    @transaction.atomic(durable=True)
```

Comment:
Any specific reason for this?

Reply:
Only the fact that you used it. It has something to do with being in subsequent transaction contexts and ensuring that each context is "atomic", not only transaction as a whole, right? Seems like it increases security of the solution.

Reply:
I used it because my code sends acknowledgement request to Google (i.e. we notify google that user consumed the product), and thus I don't want to appear in a situation where some outer transaction rolls back my actions while I already sent the acknowledgement.
Well I guess it doesn't harm your code unless you can imagine some case when there is some outer transaction which wraps your code.

Reply:
You're right. It's not used in apple case. Even if apple needs to know about consumables, request to confirm actual usage comes via a notification. Removed.